cicecore/cicedynB/infrastructure/ice_grid.F90

```diff
@@ -1727,15 +2051,32 @@ subroutine makemask
          jhi = this_block%jhi
 
          ! needs to cover halo (no halo update for logicals)
-         tmask(:,:,iblk) = .false.
-         umask(:,:,iblk) = .false.
+         tmask(:,:,iblk)   = .false.
+         umask(:,:,iblk)   = .false.
+         umaskCD(:,:,iblk) = .false.
+         nmask(:,:,iblk)   = .false.
+         emask(:,:,iblk)   = .false.
          do j = jlo-nghost, jhi+nghost
          do i = ilo-nghost, ihi+nghost
-            if ( hm(i,j,iblk) > p5) tmask(i,j,iblk) = .true.
-            if (uvm(i,j,iblk) > p5) umask(i,j,iblk) = .true.
+            if ( hm(i,j,iblk)   > p5  ) tmask  (i,j,iblk)   = .true.
+            if (uvm(i,j,iblk)   > p5  ) umask  (i,j,iblk)   = .true.
+            if (uvmCD(i,j,iblk) > c1p5) umaskCD(i,j,iblk)   = .true.
+            if (npm(i,j,iblk)   > p5  ) nmask  (i,j,iblk)   = .true.
+            if (epm(i,j,iblk)   > p5  ) emask  (i,j,iblk)   = .true.
          enddo
          enddo
 
+      enddo
+      !$OMP END PARALLEL DO
+
+      !$OMP PARALLEL DO PRIVATE(iblk,i,j,ilo,ihi,jlo,jhi,this_block)
+      do iblk = 1, nblocks
```

Comment:
why was this loop split from the one above?

Reply:
I think this may have generated different results.  Would need to check again.